cmscontrib/PrometheusExporter.py

```diff
@@ -0,0 +1,293 @@
+        self.export_workers = not args.no_workers
+        self.export_queue = not args.no_queue
+        self.export_communiactions = not args.no_communications
+        self.export_users = not args.no_users
+        self.evaluation_service = self.connect_to(
+            ServiceCoord("EvaluationService", 0))
+
+    def collect(self):
+        with SessionGen() as session:
+            if self.export_subs:
+                yield from self.collect_subs(session)
+            if self.export_workers:
+                yield from self.collect_workers()
+            if self.export_queue:
+                yield from self.collect_queue()
+            if self.export_communiactions:
+                yield from self.collect_communications(session)
+            if self.export_users:
+                yield from self.collect_users(session)
+
+    def collect_subs(self, session):
+        # compiling / max_compilations / compilation_fail / evaluating /
+        # max_evaluations / scoring / scored / total
+        stats = AdminWebServer.submissions_status(self.contest_id)
+        metric = GaugeMetricFamily(
+            "cms_submissions",
+            "Number of submissions per category",
+            labels=["status"],
+        )
+        for status, count in stats.items():
+            metric.add_metric([status], count)
+        yield metric
+
+        metric = CounterMetricFamily(
+            "cms_task_submissions",
+            "Number of submissions per task",
+            labels=["task"],
+        )
+        data = session.query(Task.name, func.count(Submission.id)) \
+            .select_from(Submission) \
+            .join(Task)
+        if self.contest_id is not None:
+            data = data.filter(Task.contest_id == self.contest_id)
+        data = data \
+            .group_by(Task.name) \
+            .all()
+        for task_name, count in data:
+            metric.add_metric([task_name], count)
+        yield metric
+
+        metric = CounterMetricFamily(
+            "cms_submissions_language",
+            "Number of submissions per language",
+            labels=["language"],
+        )
+        data = session.query(Submission.language, func.count(Submission.id)) \
+            .select_from(Submission) \
+            .group_by(Submission.language)
+        if self.contest_id is not None:
+            data = data.join(Task) \
+                .filter(Task.contest_id == self.contest_id)
+        data = data.all()
+        for language, count in data:
+            metric.add_metric([language], count)
+        yield metric
+
+    def collect_workers(self):
+        status = self.evaluation_service.workers_status().get()
+        metric = GaugeMetricFamily(
+            "cms_workers",
+            "Number of cmsWorker instances",
+            labels=["status"],
+        )
+        metric.add_metric(["total"], len(status))
+
+        connected = len([1 for worker in status.values() if worker["connected"]])
```

Comment:
You can use generator expressions to avoid constructing the full list: `sum(1 for … in … if …)`. Similarly below.